### PR TITLE
New test case submission for PTX-15696

### PR DESCRIPTION
**Pull request by vprabhakar-px**
STEP: validate fio-writes app's volume fio-log-fio-5 has been deleted in the volume driver
2023-03-24 17:21:54 +0530:[INFO] [{ReplResyncOnPoolExpand}] validate fio-writes app's volume fio-log-fio-5 has been deleted in the volume driver
INFO[2023-03-24 17:21:54] Verifying : Description : fio-writes's volume fio-log-fio-5 deleted successfully? 
INFO[2023-03-24 17:21:54] Actual:true, Expected: true                  
STEP: destroy the fio-writes app's volumes
2023-03-24 17:21:54 +0530:[INFO] [{ReplResyncOnPoolExpand}] [tests.DeleteVolumes.func1:#1393] - destroy the fio-writes app's volumes
2023-03-24 17:21:55 +0530:[INFO] [{ReplResyncOnPoolExpand}] [k8s.(*K8s).DeleteVolumes:#3114] - [fio-writes] Storage class is not found: fio-sc, skipping deletion
2023-03-24 17:21:55 +0530:[INFO] [{ReplResyncOnPoolExpand}] [k8s.(*K8s).DeleteVolumes:#3114] - [fio-writes] Storage class is not found: fio-log, skipping deletion
2023-03-24 17:21:55 +0530:[INFO] [{ReplResyncOnPoolExpand}] [k8s.(*K8s).DeleteVolumes:#3227] - [fio-writes] Destroyed PVCs for StatefulSet: fio
INFO[2023-03-24 17:21:56] Verifying : Description : Test completed successfully ? 
INFO[2023-03-24 17:21:56] Actual:PASS, Expected: PASS         
INFO[2023-03-24 17:21:56] --------Test End------                       
INFO[2023-03-24 17:21:56] #Test: ReplResyncOnPoolExpand                
INFO[2023-03-24 17:21:56] #Description: Resync failed for a volume after pool came out of rebalance  
INFO[2023-03-24 17:21:56] ------------------------                     

• [SLOW TEST:9479.667 seconds]
{ReplResyncOnPoolExpand}
/Users/vprabhakar/Desktop/work/torpedo/torpedo/tests/basic/storage_pool_test.go:8364
  Resync volume after rebalance
  /Users/vprabhakar/Desktop/work/torpedo/torpedo/tests/basic/storage_pool_test.go:8387
------------------------------

### CODE REVIEW

File: tests/basic/storage_pool_test.go [hunk 1]
```diff
@@ -8361,3 +8361,124 @@ var _ = Describe("{AddDiskAddDriveAndDeleteInstance}", func() {
...
+		defer revertReplica()
+		for _, eachVol := range volumes {
+			getReplicaSets, err := Inst().V.GetReplicaSets(eachVol)
+			log.FailOnError(err, "Failed to get replication factor on the volume")
+			volumeReplicaMap[eachVol.Name] = len(getReplicaSets[0].Nodes)
+
+			if len(getReplicaSets[0].Nodes) != 2 {
+				err := Inst().V.SetReplicationFactor(eachVol, 2, nil, nil, true)
+				if err != nil {
+					re := regexp.MustCompile("Failed to update volume: No change requested")
```
[the remark this answers is not in the document]
Comment (vprabhakar-px) [reply tests/basic/storage_pool_test.go:8435]: fixed

File: tests/basic/storage_pool_test.go [hunk 2]
```diff
@@ -8361,3 +8361,124 @@ var _ = Describe("{AddDiskAddDriveAndDeleteInstance}", func() {
...
+		expectedStatus := "Online"
+		err = WaitForPoolStatusToUpdate(*nodeDetail, expectedStatus)
+		log.FailOnError(err, fmt.Sprintf("node %s pools are not in status %s", nodeDetail.Name, expectedStatus))
+		for _, eachVol := range volumes {
+			volumeInspect, err := Inst().V.InspectVolume(eachVol.ID)
+			log.FailOnError(err, fmt.Sprintf("Failed to get details of volume [%v]", eachVol.Name))
+
+			if fmt.Sprintf("[%v]", volumeInspect.Status) != "VOLUME_STATUS_UP" {
+
+				fmt.Printf("Volume Details [%v] is [%v]", volumeInspect.Status, volumeInspect.State)
```
[the remark this answers is not in the document]
Comment (vprabhakar-px) [reply tests/basic/storage_pool_test.go:8472]: done

File: tests/basic/storage_pool_test.go [hunk 3]
```diff
@@ -8483,3 +8500,120 @@ var _ = Describe("{DriveAddAsJournal}", func() {
...
+
+		defer revertReplica()
+		for _, eachVol := range volumes {
+			getReplicaSets, err := Inst().V.GetReplicaSets(eachVol)
+			log.FailOnError(err, "Failed to get replication factor on the volume")
+			volumeReplicaMap[eachVol.Name] = len(getReplicaSets[0].Nodes)
+
+			if len(getReplicaSets[0].Nodes) != 2 {
+				err := Inst().V.SetReplicationFactor(eachVol, 2, nil, nil, true)
+				if err != nil {
```
Comment (lsrinivas-pure) [inline tests/basic/storage_pool_test.go:8574]: if is not required here

**Review by lsrinivas-pure — APPROVED** [review]
lgtm, apart from minor comment

File: tests/basic/storage_pool_test.go (hunk 3 above)
Comment (vprabhakar-px) [reply tests/basic/storage_pool_test.go:8574]: oops I missed it updating ! will fix it along with other review that I send for review

**Pull request merged**